### DISCUSSION TITLE
oAuth how to get a users name (and possibly other variables) from provider

### DISCUSSION THREAD

**Discussion by LickABrick** [pocketbase/pocketbase#1592] [Q&A] [answered]
Hi, is it possible to automatically fill the users name with the name they have set at the provider? E.g. in Microsoft my name is 'John Doe [Company Name]' it will also be this in PB.

The following might just scope to my specific project, but Microsoft also has an 'TenantID' which I would like to use in my project but currently I see no way to get this.

**Comment by ganigeorgiev** [comment, marked as answer]
The raw user OAuth2 fields are returned under the `meta.rawUser` response object.

It is not possible to autofill the provider specific fields at the moment in a single request (_without extending PocketBase and using the [`OnRecordAuthRequest` hook](https://pocketbase.io/docs/event-hooks/#record-api-hooks)_).
You'll have to send a second `update()` user request with the previously extracted OAuth2 fields you want to assign.

**Comment by LickABrick** [comment]
Ah I see, Thanks. I'm kinda new to Javascript so not sure what the best way to do this is. Is it ok to do a .then after .authWithOAuth2 in which I update the fields?

**Reply by ganigeorgiev** [reply]
`authWithOAuth2()` returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) so you need to resolve/await it first (yes, you can chain it in a `then` callback).

Here is an example using `await`:

```js
// sign-in/sign-up
const authData = await pb.collection('users').authWithOAuth2(
  'google',
  'CODE',
  'VERIFIER',
  'REDIRECT_URL',
);

console.log(authData.meta.rawUser);

// update
await pb.collection('users').update(authData.record.id, {
  someField: authData.meta.rawUser.YOUR_PROVIDER_FIELD,
});
```

**Reply by LickABrick** [reply]
This is what I needed. Thanks for the really fast response!! Love the project ❤️
